ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuation.kt

```diff
@@ -0,0 +1,113 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.rule.engine.core.api.ElementType
+import com.pinterest.ktlint.rule.engine.core.api.Rule
+import com.pinterest.ktlint.rule.engine.core.api.firstChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.hasNewLineInClosedRange
+import com.pinterest.ktlint.rule.engine.core.api.indent
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpace
+import com.pinterest.ktlint.rule.engine.core.api.lastChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.leavesIncludingSelf
+import com.pinterest.ktlint.rule.engine.core.api.nextCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.noNewLineInOpenRange
+import com.pinterest.ktlint.rule.engine.core.api.prevCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.prevLeaf
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceAfterMe
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceBeforeMe
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+/**
+ * When present, align chain operators('.' or '?.') at previous line with next method call
+ */
+public class ChainMethodContinuation :
+    StandardRule("chain-method-continuation"),
+    Rule.Experimental {
+    private val chainOperators =
+        TokenSet.create(
+            ElementType.DOT_QUALIFIED_EXPRESSION,
+            ElementType.SAFE_ACCESS_EXPRESSION,
+        )
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        node.takeIf {
+            it.elementType in chainOperators
+        }?.let {
+            visitChains(node, autoCorrect, emit)
+        }
+    }
+
+    private fun visitChains(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
```

Comment:
`you can just call chainOperator.isPrecededByMultilineChain() which is more expressive than using a variable` Indeed calling the extension method will be more expensive as we have to calculate it 3 times. Are you still suggesting it to move it to extension method?

Reply:
Actually, it will not be called more than twice because it is used in mutually exclusive branches of the if-else. I value code readability over premature code optimazations. If would be different if the function would be called inside a loop which is executed tens or hundreds of times.